### PR TITLE
Data Quality Issues Tool & Handle deletion of Issues

### LINKED ISSUE #8: data quality exceptions
after #1/#2 - to make these less noisy it should be easy to add data quality exceptions for 'known' issues

likely to differ from how we did things in billy since we've learned a lot about what exceptions we encounter

### LINKED ISSUE #52: Handle the deletion of data quality issues
Spun off from @hiteshgarg14's thoughts in this comment: https://github.com/openstates/admintools/pull/43#issuecomment-317282434

@hiteshgarg14, does this happen just for name resolution, or for all data quality issues that are fixed by admin edits and/or patches?

@jamesturk, do you think we should delete _relevant_ data quality issues each time a fix is made? Or should we leave the counts as-is until the full `manage.py import all` is run again?

Personally, I'm in favor of having an AWS task that regularly performs the `import all`, or having a button in the nav bar to trigger an AWS task to do so (and otherwise it wouldn't happen besides on-pupa-import). That way we wouldn't need to keep track of which DQ issues have been solved by any given admin action or patch.

### LINKED ISSUE #55: avoid hard-coded path /admintools/
instead of having /admintools/ be a hard-coded part of the URL let's have the app live wherever the upper-level urls.py mounts it.  in practice I think this will mean removing /admintools/ from urls.py (and then when including this in a Django project it would be included like ``` url(r'^admintools/', include('admintools.urls')),```

### LINKED ISSUE #56: issue count mismatch
was playing w/ the admin today and noticed the numbers don't seem to match up when I go between the overview & the jurisdiction page

one:
nh
two:
nh2

this confused me for a while but I realized that it is because Organization is split out into Membership/Post/etc.   -- we should consider:
- splitting these out on the first view
- combining the sections on the second view
- other options?

### LINKED ISSUE #57: fix ordering of posts
to more easily find what someone is looking for, should be sorted
post-order

### LINKED ISSUE #58: retirement tool should be linked from 'too many members' view
it'd be great if you could start the retirement process from the 'too many members' view instead of separately going to the retirement view to start to fix it

### LINKED ISSUE #60: minor visual tweaks
some really minor things

- [ ] in a lot of places ordered lists are used for things that don't really have a numeric order <ul> is probably a better choice in these places 

uls

- [ ] headers shouldn't be clickable if they aren't links anymore
header-links

### LINKED ISSUE #61: navigation improvements
I think it'd make sense to have the top nav bar add links for retirement & name resolution tool

### LINKED ISSUE #63: clarify (or remove) 'all' parameter to import management command
right now this parameter is required via 

```parser.add_argument('all', nargs='+', type=str)```

but it doesn't seem to be used

### PR DESCRIPTION
**Pull request by hiteshgarg14**
Introduced two new fields:-
 - status => active/ignored
 - message => for ignored issues

1. From now on `Status Page` and `Data Quality Issues Section` only `active` issues will be displayed.
2. The user can remove the `ignored` issues.
2. Coverage is 97%

closes #52 
closes #8 
closes #60 
closes #61 
closes #63 
closes #55
closes #56
closes #57 
closes #58

### CODE REVIEW

File: admintools/importers/orgs.py [hunk 1]
```diff
@@ -24,6 +26,12 @@ def create_org_issues(queryset, issue, jur):
 def orgs_issues():
     all_jurs = Jurisdiction.objects.order_by('name')
     for jur in all_jurs:
+        DataQualityIssue.objects.filter(jurisdiction=jur, status='active'
+                                        ).filter(
+                                        Q(issue__startswith='organization-')
```
Comment (jamesturk) [inline admintools/importers/orgs.py]: instead of these startswith queries could you use the content_type values?

File: admintools/views.py [hunk 2]
```diff
@@ -99,30 +111,42 @@ def overview(request):
...
     issues = IssueType.choices()
     for issue, description in issues:
         related_class = IssueType.class_for(issue)
         cards[related_class][issue] = {}
+        exceptions[related_class][issue] = {}
         issue_type = IssueType.class_for(issue) + '-' + issue
         alert = IssueType.level_for(issue)
         cards[related_class][issue]['alert'] = True if alert == 'error' \
             else False
+        exceptions[related_class][issue]['alert'] = True if alert == 'error' \
```
Comment (jamesturk) [inline admintools/views.py]: probably . more clearly written as 

```
exceptions[related_class][issue]['alert'] = (alert == 'error')

File: admintools/migrations/0007_dataqualityissue_message.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('admintools', '0006_dataqualityissue_status'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='dataqualityissue',
+            name='message',
+            field=models.CharField(blank=True, max_length=5000),
```
Comment (jamesturk) [inline admintools/migrations/0007_dataqualityissue_message.py]: IMO TextField() is probably better than max_length=5000

File: admintools/importers/bills.py [hunk 4]
```diff
@@ -11,10 +11,12 @@ def create_bill_issues(queryset, issue, jur):
     for query_obj in queryset:
         if not DataQualityIssue.objects.filter(object_id=query_obj.id,
```
Comment (jamesturk) [inline admintools/importers/bills.py]: doing this query within the for loop is going to be really expensive (imagine a situation w/ 5000 objects in the queryset- this check is done that many times by round-trips to the DB)

I think it'd be a lot more efficient if you could gather a list of all of the ignored issues for this (jurisdiction+issue_type) or something and then do a comparison to see if the object_id is present in the pre-selected list

**Comment by mileswwatkins** [comment]
Regarding the documentation, the technical part regarding views and helpers looks thorough!

Could you write up a paragraph at the very top, summarizing the tool's purpose (and include links to your blog posts, for additional reference)? And also, can you include examples of how you'd run the two management commands, including how you'd need to point to the DB, or what the structure of the input patch file is (feel free to just point to the test directory's patch-file example).

**Comment by hiteshgarg14** [comment]
>And also, can you include examples of how you'd run the two management commands, including how you'd need to point to the DB

Do you mean starting from how to setup this Django-app to run management commands?

**Comment by mileswwatkins** [comment]
No, I think more to answer basic questions about what these management commands do, and what they take as input:

> Import Data Quality Issues into Database

- … from where?
- what table are the imported issues stored in?
- what happens to existing data quality issues that are already in the database?

> Import People Related Issue Resolver Patches into Database

- what's an example of the input format?
- what table are the imported patches stored in?
- what happens to existing patches that are already in the database?

**Pull request merged**